robots/norbot2.h

```diff
@@ -0,0 +1,99 @@
+
+#pragma once
+
+// BoB robotics includes
+#include "../common/i2c_interface.h"
+
+
+// Standard C includes
+#include <cmath>
+#include <cstdint>
+
+// Standard C++ includes
+#include <vector>
+
+// third party includes
+#include "../third_party/units.h"
+
+
+//----------------------------------------------------------------------------
+
+//----------------------------------------------------------------------------
+//! An interface for 4 wheeled, Arduino-based robots developed at the University of Sussex
+class Norbot2 
+{
+    using meters_per_second_t = units::velocity::meters_per_second_t;
+    using millimeter_t = units::length::millimeter_t;
+    
+
+public:
+    Norbot2(const char *path = "/dev/i2c-1", int slaveAddress = 0x29)
+      : m_I2C(path, slaveAddress)
+      , m_forwardVelocity(0.0f)
+      , m_turningAngle(0.0f)
+    {}
+
+    // destructor		
+    ~Norbot2() 
+    {
+        stopMoving();
+    }
+
+    void move(uint8_t speed, uint8_t turningAngle)
```

Comment:
For consistancy with the other robots, the speed should be a normalised ``float`` i.e. between -1 and 1 and the steering should either be in the same form or as a ``degree_t`` if it actually corresponds to setting an absolute angle

Reply:
:+1: 

I'd also check that the user has passed valid arguments:
```c++
BOB_ASSERT(speed >= -1.f && speed <= 1.f);
BOB_ASSERT(turningAngle >= -45_deg && turningAngle <= 45_deg);
```